pkg/mssqlcontainer/ingest/ingest.go

```diff
@@ -0,0 +1,135 @@
+package ingest
+
+import (
+	"fmt"
+	"github.com/microsoft/go-sqlcmd/internal/container"
+	"github.com/microsoft/go-sqlcmd/internal/uri"
+	"github.com/microsoft/go-sqlcmd/pkg/mssqlcontainer/ingest/extract"
+	"github.com/microsoft/go-sqlcmd/pkg/mssqlcontainer/ingest/location"
+	"github.com/microsoft/go-sqlcmd/pkg/mssqlcontainer/ingest/mechanism"
+	"path/filepath"
+	"strings"
+)
+
+type ingest struct {
+	uri         uri.Uri
+	location    location.Location
+	controller  *container.Controller
+	mechanism   mechanism.Mechanism
+	options     mechanism.BringOnlineOptions
+	extractor   extract.Extractor
+	containerId string
+	query       func(text string)
+}
+
+func (i *ingest) IsExtractionNeeded() bool {
+	i.extractor = extract.NewExtractor(i.uri.FileExtension(), i.controller)
+	if i.extractor == nil {
+		return false
+	} else {
+		return true
+	}
+}
+
+func (i *ingest) IsRemoteUrl() bool {
+	return !i.location.IsLocal()
+}
+
+func (i *ingest) UrlFilename() string {
+	return i.uri.Filename()
+}
+
+func (i *ingest) IsValidScheme() bool {
+	for _, s := range i.location.ValidSchemes() {
+		if s == i.uri.Scheme() {
+			return true
+		}
+	}
+	return false
+}
+
+func (i *ingest) CopyToContainer(containerId string) {
+	destFolder := "/var/opt/mssql/backup"
+	if i.mechanism != nil {
+		destFolder = i.mechanism.CopyToLocation()
+	}
+	if i.location == nil {
+		panic("location is nil, did you call NewIngest()?")
+	}
+
+	i.containerId = containerId
+	i.location.CopyToContainer(containerId, destFolder)
+	i.options.Filename = i.uri.Filename()
+
+	if i.options.Filename == "" {
+		panic("filename is empty")
```

Comment:
>ename is empty") [](http://example.com/codeflow?start=12&length=16)

if the user provides a mechanism but the URL doesn't have a detectable file name, can we just generate a unique name with an extension that matches the mechanism?